src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -844,6 +965,12 @@ SyntaxAnalyzerResultPtr SyntaxAnalyzer::analyzeSelect(
         /// Remove duplicate items from ORDER BY.
         optimizeOrderBy(select_query);
 
+        /// Remove duplicate ORDER BY from subqueries.
+        optimizeDuplicateOrderBy(query, context);
```

Comment:
But for subquery that querying distributed table, removing `ORDER BY` expression is not correct, since this will move all the work from shards to the initial node, right?

Reply:
Can you give an example of a distributed table? Also, I will try to fix other remarks soon.

Reply:
[Distributed engine](https://clickhouse.tech/docs/en/engines/table_engines/special/distributed/) or you can use [`remote`](https://clickhouse.tech/docs/en/sql_reference/table_functions/remote/)/`cluster` functions (that will create temporary distributed table)

So something like this (see `distributed_group_by_no_merge` tricky cases)

```
CREATE TABLE numbers AS system.numbers;
INSERT INTO numbers SELECT * FROM numbers(1000000);

SELECT DISTINCT number
FROM 
(
    SELECT number
    FROM remote('127.{2,3}', currentDatabase(), numbers)
    LIMIT 1
)
```

Actually maybe it will be a good idea to disable this optimization if it has distributed table inside

Reply:
Thanks!